routers/web/repo/view.go

```diff
@@ -234,7 +234,7 @@ func getFileReader(ctx gocontext.Context, repoID int64, blob *git.Blob) ([]byte,
 	}
 
 	meta, err := git_model.GetLFSMetaObjectByOid(ctx, repoID, pointer.Oid)
-	if err != nil && err != git_model.ErrLFSObjectNotExist { // fallback to plain file
+	if err != nil && err == git_model.ErrLFSObjectNotExist { // fallback to plain file
```

Comment:
I don't think the logic is right. Maybe you mean.
```suggestion
	if err != nil  { // fallback to plain file
```

Reply:
That's how I initially structured it, but upon rereading the logic, the bare `err != nil` check[ happens a couple of lines later](https://github.com/go-gitea/gitea/blob/d3970dd2cd9cf6788621ef58f6b331cd6ef5d5d7/routers/web/repo/view.go#L242). I suspect it's this way to allow closing the file in one spot instead of spreading it throughout error handling logic.

Reply:
Yes, and line 242-244 should be removed.

Reply:
Fixed in https://github.com/go-gitea/gitea/pull/31812/commits/3ac3418eb3542fe39a2a0c0d6dfba39fe0371fcb